api/src/main/java/org/pmiops/workbench/db/dao/BillingProjectBufferEntryDao.java

```diff
@@ -0,0 +1,16 @@
+package org.pmiops.workbench.db.dao;
+
+import org.pmiops.workbench.db.model.BillingProjectBufferEntry;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface BillingProjectBufferEntryDao  extends CrudRepository<BillingProjectBufferEntry, Long> {
+
+  BillingProjectBufferEntry findByFireCloudProjectName(String fireCloudProjectName);
+
+  @Query("SELECT COUNT(*) FROM BillingProjectBufferEntry WHERE status=0 OR status=2 OR status=3")
```

Comment:
I'd not include `3` here. I may have said differently in my doc - I don't remember.

If it's actively assigning, it's been claimed (either for assignment, or if that fails for deletion), so I think it's correct to exclude it from the buffer size.